core/esmf-aspect-meta-model-java/src/main/java/org/eclipse/esmf/aspectmodel/loader/AspectModelLoader.java

```diff
@@ -173,6 +178,37 @@ public AspectModel load( final InputStream inputStream ) {
       return buildAspectModel( loaderContext.loadedFiles() );
    }
 
+   public AspectModel loadFromArchive( final String pathToArchive ) {
+      File zipFile = new File( pathToArchive );
+
+      if ( !zipFile.exists() || !zipFile.isFile() ) {
+         throw new RuntimeException( new FileNotFoundException( "The specified file does not exist or is not a file." ) );
+      }
+
+      List<AspectModelFile> aspectModelFiles = new ArrayList<>();
+
+      try ( ZipFile zip = new ZipFile( zipFile ) ) {
+         Enumeration<? extends ZipEntry> entries = zip.entries();
+
+         while ( entries.hasMoreElements() ) {
+            ZipEntry entry = entries.nextElement();
+
+            if ( entry.getName().endsWith( ".ttl" ) ) {
+               try ( InputStream inputStream = zip.getInputStream( entry ) ) {
+                  AspectModelFile aspectModelFile = AspectModelFileLoader.load( inputStream );
+                  aspectModelFiles.add( aspectModelFile );
+               } catch ( IOException e ) {
+                  LOG.error( String.format( "Error reading entry: %s - %s", entry.getName(), e.getMessage() ) );
+               }
+            }
+         }
+      } catch ( IOException e ) {
+         LOG.error( String.format( "Error reading the ZIP file: %s", e.getMessage() ) );
+      }
+
+      return buildAspectModel( aspectModelFiles );
```

Comment:
Here another important step is missing: Analoguous to `load( Collection<File> )`, before calling `buildAspectModel()` you need to call `resolve()`. We expect that the namespace package does not contain references to model elements that are not defined in it, but it's not mandatory, so resolution must still be done.